pulp_2_tests/tests/rpm/api_v2/test_modularity.py

```diff
@@ -55,3 +73,22 @@ def test_sync_publish_repo(self):
             'modules',
             api.safe_handler,
         )
+
+    @skip_if(bool, 'repos', False)
+    def test_02_sync_copy_repo(self):
+        """Test sync and copy modular RPM repository."""
+        api.Client(self.cfg).post(
+            urljoin(self.repos[1]['_href'], 'actions/associate/'),
```

Comment:
Pulp will be able copy `modulemd` and  `modulemd_defaults`. Basically we will be able to copy `moduleA` from `repoA` to `repoB`.

```python

      self.client.post(urljoin(repos[1]['_href'], 'actions/associate/'), {
            'source_repo_id': repos[0]['id'],
            'override_config': {'recursive': recursive},
            'criteria': {
                'filters': {'unit': {'name':'walrus', 'stream':'0.71'}},
                'type_ids': ['modulemd'],
            },
        })
``` 
The main goal behind this test is to confirm that `modulesmd` or `modulemd_defaults` can be copied, and when `recursive` flag is set to True the RPM packages associated with those `modules` will be copied as well.